source/common/sequencer_impl.cc

```diff
@@ -81,6 +81,9 @@ void SequencerImpl::updateStartBlockingTimeIfNeeded() {
 
 void SequencerImpl::run(bool from_periodic_timer) {
   ASSERT(running_);
+  // Update time so that our time source will yield an up-to-date monotonic time sample to work with
+  // during this function call.
+  dispatcher_.updateApproximateMonotonicTime();
```

Comment:
if I understand this correctly, then real monotonicTime() is called in updateApproximateMonotonicTime, which means no optimization gain for this specific call. 
Can you please call out where the cached_time_source..monotonicTime() would be saving time.

Reply:
I added a more elaborate comment in 823c15eba8d6261edb832840479e1c69bdd1d7e8, does that look helpful?

Reply:
Very helpful. Thanks.